### PR TITLE
feat(udf): allow remote UDF registeration

### PR DESCRIPTION
**Pull request by shinyano**
# New feature:
User can register UDFs using local python scripts into iginx that runs on a different host.

# Usage:
- client
Use normal sql. Session will decide whether iginx service is on a different host based on connection parameters.
- session
Use `executeRegisterTask(sql)` method.
**NOTES**: 
Currently session provides `executeRegisterTask(sql, isRemote)` to perform loading remote scripts behaviour, ignoring whether iginx is really on a different host. This is for testing only, and can be optimized if we find a better way to do tests.
Also, `executesql` stays functional when local absolute script path is provided in original statement. Local means scripts and iginx are on the same host.

### CODE REVIEW

File: test/src/test/java/cn/edu/tsinghua/iginx/integration/func/udf/UDFIT.java [hunk 1]
```diff
@@ -170,6 +182,51 @@ public void dropTasks() {
     taskToBeRemoved.clear();
   }
 
+  private void executeReg(String statement, boolean isRemote) {
```
Comment (SolomonAnn) [inline test/src/test/java/cn/edu/tsinghua/iginx/integration/func/udf/UDFIT.java:185]: 在单机测试下，该函数isRemote传入true是否有意义？

File: test/src/test/java/cn/edu/tsinghua/iginx/integration/func/udf/UDFIT.java [hunk 2]
```diff
@@ -1353,4 +1421,8 @@ public void testModuleInstallFail() {
       }
     }
   }
+
+  // register local UDF script into iginx that runs on different host
+  @Test
+  public void testRemoteFileReg() {}
```
Comment (SolomonAnn) [inline test/src/test/java/cn/edu/tsinghua/iginx/integration/func/udf/UDFIT.java:1427]: 补充关于remote的测试，需保证client和server在不同节点上

**Review by zhuyuqing — CHANGES_REQUESTED** [review]
remote tests can be and should be added.

File: client/src/assembly/resources/sbin/start_cli.sh [hunk 3]
```diff
@@ -72,6 +72,8 @@ case "${PARAMETERS[@]}" in
 *) PARAMETERS=("-h" "127.0.0.1" "${PARAMETERS[@]}") ;;
 esac
 
+echo ${PARAMETERS[@]}
```
Comment (SolomonAnn) [inline client/src/assembly/resources/sbin/start_cli.sh:75]: 这是为了调试打印的日志吧

File: .github/scripts/test/cli/test_remote_udf.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ls docker/client/data
+ls docker/client/data/udf
+
+set os=$1
+echo "$os"
+
+export MSYS_NO_PATHCONV=1
+# MSYS_NO_PATHCONV=1 : not to convert docker script path to git bash path
+SCRIPT_PREFIX="docker exec iginx-client /iginx_client/sbin/start_cli.sh -h host.docker.internal -e"
+if [[ $os =~ "mac" ]]; then
```
Comment (SolomonAnn) [inline .github/scripts/test/cli/test_remote_udf.sh:16]: 这个脚本就是linux专用的吧，这里是不是应该删除

File: .github/workflows/remote-test.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+              uses: ./.github/actions/iginxRunner
+              with:
+                  version: ${VERSION}
+
+            - name: Register Remote UDFs
+              shell: bash
+              run: |
+                  if [ "$RUNNER_OS" == "Linux" ]; then
+                    chmod +x "${GITHUB_WORKSPACE}/.github/scripts/test/cli/test_remote_udf.sh"
+                    "${GITHUB_WORKSPACE}/.github/scripts/test/cli/test_remote_udf.sh" ${{ matrix.os }}
```
Comment (SolomonAnn) [inline .github/workflows/remote-test.yml:105]: 无需传入os

**Pull request merged**